File: java/src/examples/simple/Box.java
package examples.simple;

public class Box<T> {
    private T value;

    public Box(T value) {
        this.value = value;
    }

    public T getValue() {
        return value;
    }

    public static void main(String[] args) {
        /**
         * Box example
         */
        Box<Integer> wrapped = new Box<>(10);
        Box<String> wrappedString = new Box<>("abcd");

        System.out.println(wrapped.getValue());
        System.out.println(wrappedString.getValue());
    }
}

File: cs/GenericsComparison/Simple/Box.cs
using System;

namespace Example.simple
{
    public class Box<T> {
        public T Value { get; }

        public Box(T value) => Value = value;
    }

    public class Program
    {
        public static void Main(string[] args) {
            var wrapped = new Box<int>(10);
            var wrappedString = new Box<string>("abcd");

            Console.Out.Write(wrapped.Value);
            Console.Out.Write(wrappedString.Value);
        }
    }
}